pandas/core/generic.py

```diff
@@ -8532,7 +8533,7 @@ def compare(
         else:
             axis = self._get_axis_number(align_axis)
 
-        diff = concat([self, other], axis=axis, keys=keys)
+        diff = concat([self, other], axis=axis, keys=suffixes)
```

Comment:
these must be a tuple and non-None right? can you add a check and tests the cases (e.g. this is similar to what we checkin merge for example)